src/main/io/vtx_rtc6705.c

```diff
@@ -0,0 +1,325 @@
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Created by jflyper */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "platform.h"
+
+#if defined(VTX_RTC6705) && defined(VTX_CONTROL)
+#include "build/build_config.h"
+
+#include "cms/cms.h"
+#include "cms/cms_types.h"
+
+#include "common/printf.h"
+#include "common/utils.h"
+
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+#include "drivers/system.h"
+#include "drivers/vtx_common.h"
+#include "drivers/vtx_rtc6705.h"
+
+#include "fc/rc_controls.h"
+#include "fc/runtime_config.h"
+
+#include "io/vtx_rtc6705.h"
+#include "io/vtx_string.h"
+
+#include "build/debug.h"
+
+bool canUpdateVTX(void);
+
+PG_REGISTER_WITH_RESET_TEMPLATE(vtxRTC6705Config_t, vtxRTC6705Config, PG_VTX_RTC6705_CONFIG, 0);
+
+PG_RESET_TEMPLATE(vtxRTC6705Config_t, vtxRTC6705Config,
+    .band = 4,    //Fatshark/Airwaves
+    .channel = 1, //CH1
+    .power = VTX_RTC6705_DEFAULT_POWER
+);
+
+#define WAIT_FOR_VTX while (!canUpdateVTX()) {}
+
+#if defined(CMS) || defined(VTX_COMMON)
+#ifdef RTC6705_POWER_PIN
+static const char * const rtc6705PowerNames[RTC6705_POWER_COUNT] = {
+    "---", "25 ", "200",
+};
+#else
+static const char * const rtc6705PowerNames[RTC6705_POWER_COUNT] = {
+    "25 ", "200",
+};
+#endif
+#endif
+
+#ifdef VTX_COMMON
+static vtxVTable_t rtc6705VTable;    // Forward
+static vtxDevice_t vtxRTC6705 = {
+    .vTable = &rtc6705VTable,
+    .capability.bandCount = 5,
+    .capability.channelCount = 8,
+    .capability.powerCount = RTC6705_POWER_COUNT,
+    .bandNames = (char **)vtx58BandNames,
+    .chanNames = (char **)vtx58ChannelNames,
+    .powerNames = (char **)rtc6705PowerNames,
+};
+#endif
+
+typedef struct rtc6705Dev_s {
+    uint8_t band;    // 1-8, 1-based
+    uint8_t channel; // 1-8, 1-based
+    uint8_t power; // 0-(RTC6705_POWER_COUNT-1)
+} rtc6705Dev_t;
+
+rtc6705Dev_t rtc6705Dev;
+
+
+bool vtxRTC6705Init(void)
+{
+    memset(&rtc6705Dev, 0, sizeof(rtc6705Dev));
+
+    vtxRTC6705.vTable = &rtc6705VTable;
+    vtxCommonRegisterDevice(&vtxRTC6705);
+
+    return true;
+}
+
+static void vtxRTC6705Configure(void)
+{
+    rtc6705SetRFPower(rtc6705Dev.power - 1);
+    rtc6705SetChannel(rtc6705Dev.band - 1, rtc6705Dev.channel - 1);
+}
+
+#ifdef RTC6705_POWER_PIN
+static void vtxRTC6705EnableAndConfigure(void)
+{
+    WAIT_FOR_VTX;
+
+    rtc6705Enable();
+
+    delay(RTC6705_BOOT_DELAY);
+
+    vtxRTC6705Configure();
+}
+#endif
+
+void vtxRTC6705Process(uint32_t now)
+{
+    UNUSED(now);
+
+    static bool configured = false;
+    if (!configured) {
```

Comment:
again, it was following the pattern used by the tramp and smartaudio implementations.  Another for the tech-debt pile.